android-o/AutofillFramework/AutofillService/Model/FilledAutofillFieldCollection.cs

```diff
@@ -0,0 +1,265 @@
+                    }
+                }
+
+                if (IsW3CTypePrefix(hint) && nextHint != null && IsW3CTypeHint(nextHint))
+                {
+                    hint = nextHint;
+                    i++;
+                    if (CommonUtil.DEBUG)
+                        Log.Debug(CommonUtil.TAG, "Hint is a W3C type prefix; using " + hint + " instead");
+                }
+
+                if (IsW3CAddressType(hint) && nextHint != null)
+                {
+                    hint = nextHint;
+                    i++;
+                    if (CommonUtil.DEBUG)
+                        Log.Debug(CommonUtil.TAG, "Hint is a W3C address prefix; using " + hint + " instead");
+                }
+
+                // Then check if the "actual" hint is supported.
+
+
+                if (AutofillHints.IsValidHint(hint))
+                {
+                    mHintMap[hint] = filledAutofillField;
+                }
+                else
+                {
+                    Log.Error(CommonUtil.TAG, "Invalid hint: " + autofillHints[i]);
+                }
+            }
+        }
+
+        /**
+        * Populates a {@link Dataset.Builder} with appropriate values for each {@link AutofillId}
+        * in a {@code AutofillFieldMetadataCollection}.
+        *
+        * In other words, it constructs an autofill
+        * {@link Dataset.Builder} by applying saved values (from this {@code FilledAutofillFieldCollection})
+        * to Views specified in a {@code AutofillFieldMetadataCollection}, which represents the current
+        * page the user is on.
+        */
+        public bool ApplyToFields(AutofillFieldMetadataCollection autofillFieldMetadataCollection,
+            Dataset.Builder datasetBuilder)
+        {
+            bool setValueAtLeastOnce = false;
+            List<string> allHints = autofillFieldMetadataCollection.GetAllHints();
+            for (var hintIndex = 0; hintIndex < allHints.Count; hintIndex++)
+            {
+                string hint = allHints[hintIndex];
+                var fillableAutofillFields = autofillFieldMetadataCollection.GetFieldsForHint(hint);
+                if (fillableAutofillFields == null)
+                {
+                    continue;
+                }
+
+                for (var autofillFieldIndex = 0;
+                    autofillFieldIndex < fillableAutofillFields.Count;
+                    autofillFieldIndex++)
+                {
+                    var filledAutofillField = mHintMap.FirstOrDefault(x => x.Key == hint);
+                    if (filledAutofillField.Value == null)
+                    {
+                        continue;
+                    }
+
+                    var autofillFieldMetadata = fillableAutofillFields[autofillFieldIndex];
+                    AutofillId autofillId = autofillFieldMetadata.GetId();
+                    int autofillType = autofillFieldMetadata.GetAutofillType();
+                    switch (autofillType)
```

Comment:
Once the return type of `autofillFieldMetadata.GetAutofillType()` has been changed from `int` to `AutofillType` we can remove all of the casts in this switch.